### PR TITLE
New: Support automatic scraping periodically

### PR DESCRIPTION
**Pull request by brafdlog**
הוספתי אפשרות בהגדרות להגיד שרוצים שירוץ אוטומטית כל כמה שעות
זה נראה ככה:
Screenshot 2024-11-01 at 22 57 09

כשלוחצים על הפעל זה מאתחל את הטיימר של ההרצה האוטומטית.

### CODE REVIEW

File: packages/preload/src/index.ts [hunk 1]
```diff
@@ -2,4 +2,10 @@
  * @module preload
  */
 
+import { stopPeriodicScraping } from './eventsBridge';
+
 export * from './eventsBridge';
+
+// Clear the interval that scrapes periodically
+await stopPeriodicScraping();
```
Comment (baruchiro) [inline packages/preload/src/index.ts:10]: I'm not sure about this, wrote you on Whatsapp

Comment (brafdlog) [reply packages/preload/src/index.ts]: Removed this

**Comment by brafdlog** [comment]
שיניתי את המימוש שיראה את השעה של הריצה ההבאה.
לא ממשק יפה במיוחד, אבל יעשה את העבודה
Screenshot 2024-11-09 at 14 28 02

File: packages/main/src/backend/index.ts [hunk 2]
```diff
@@ -4,42 +4,71 @@ import { scrapeFinancialAccountsAndFetchTransactions } from '@/backend/import/im
...
+
+export function stopPeriodicScraping() {
+  if (intervalId) {
+    clearInterval(intervalId);
+  }
+}
 
 export async function scrapeAndUpdateOutputVendors(config: Config, optionalEventPublisher?: Events.EventPublisher) {
   const eventPublisher = optionalEventPublisher ?? new Events.BudgetTrackingEventEmitter();
+  await eventPublisher.emit(EventNames.LOG, {
```
Comment (brafdlog) [inline packages/main/src/backend/index.ts:43]: @baruchiro this is a really strange one. It seems that the first event published gets lost. I spent an hour trying to understand why the IMPORT_PROCESS_START doesn't do anything until I added an event before it, and only then it started to work. Couldn't figure out why

File: packages/main/src/handlers/index.ts [hunk 3]
```diff
@@ -67,10 +68,11 @@ export const registerHandlers = () => {
   ipcMain.on('scrape', async (event: IpcMainEvent) => {
     const config = await getConfig();
     const eventSubscriber = new BudgetTrackingEventEmitter();
-    scrapeAndUpdateOutputVendors(config, eventSubscriber);
```
Comment (brafdlog) [inline packages/main/src/handlers/index.ts:70]: @baruchiro  we had a bug here, listening to events was set up only after running scrape, and scraping was without await, so there was a race condition on whether or not the first event set would be caught by the onAny handler or not.

**Review by baruchiro — APPROVED** [review]

**Pull request merged**